samtranslator/model/preferences/deployment_preference_collection.py

```diff
@@ -152,6 +152,67 @@ def deployment_group(self, function_logical_id):
 
         return deployment_group
 
+    def _process_alarms(self, preference, group):
+        """
+        Processes the deployment preferences alarms and updates the deployment group accordingly
+
+        Parameters
+        ----------
+        preference : dict
+            Deployment preferences
+        group : dict
+            Deployment group
+
+        Raises
+        ------
+        ValueError
+            If Alarms is in the wrong format
+        """
+        if not preference.alarms or is_intrinsic_no_value(preference.alarms):
+            return
```

Comment:
what if it is in the following of: [Fn::If, condition, AWS::NoValue, someValue] and the condition may or maynot evaluates to true?

Reply:
We won't enter this condition and proceed to line 174